privacyidea/lib/tokens/radiustoken.py

```diff
@@ -137,11 +144,174 @@ def update(self, param):
+        if transaction_id is None:
+            transaction_id = options.get('state')
+
+        if transaction_id:
+            # get the challenges for this transaction ID
+            challengeobject_list = get_challenges(serial=self.token.serial,
+                                                  transaction_id=transaction_id)
+
+            for challengeobject in challengeobject_list:
+                if challengeobject.is_valid():
+                    challenge_response = True
+
+        return challenge_response
+
+    @log_with(log)
+    @check_token_locked
+    def check_challenge_response(self, user=None, passw=None, options=None):
+        """
+        This method verifies if there is a matching question for the given
+        passw and also verifies if the answer is correct.
+
+        It then returns the the otp_counter = 1
+
+        :param user: the requesting user
+        :type user: User object
+        :param passw: the password - in fact it is the answer to the question
+        :type passw: string
+        :param options: additional arguments from the request, which could
+                        be token specific. Usually "transaction_id"
+        :type options: dict
+        :return: return otp_counter. If -1, challenge does not match
+        :rtype: int
+        """
+        if options is None:
+            options = {}
+        otp_counter = -1
+
+        # fetch the transaction_id
+        transaction_id = options.get('transaction_id') or options.get('state')
+
+        # get the challenges for this transaction ID
+        if transaction_id is not None:
+            challengeobject_list = get_challenges(serial=self.token.serial,
+                                                  transaction_id=transaction_id)
+
+            for challengeobject in challengeobject_list:
+                if challengeobject.is_valid():
+                    state = challengeobject.data
+
+                    # challenge is still valid
+                    radius_response = self._check_radius(passw, options=options, radius_state=state)
+                    if radius_response == AccessAccept:
+                        # We found the matching challenge,
+                        # and the RADIUS server returned AccessAccept
+                        challengeobject.delete()
+                        otp_counter = 1
+                        break
+                    elif radius_response == AccessChallenge:
+                        # The response was valid but triggered a new challenge
+                        # Note: The second challenge currently does not work correctly
+                        # see https://github.com/privacyidea/privacyidea/issues/1792
+                        challengeobject.delete()
+                        _, _, transaction_id, _ = self.create_challenge(options=options)
+                        options["transaction_id"] = transaction_id
+                        otp_counter = 1
```

Comment:
We should return ``-1`` in this case, as the RADIUS server doesn't want to authenticate us yet

Reply:
yeah, I changes this before...